core/src/main/scala/org/broadinstitute/dsde/rawls/snapshot/WrappedSnapshot.scala

```diff
@@ -0,0 +1,23 @@
+package org.broadinstitute.dsde.rawls.snapshot
+
+import bio.terra.datarepo.model.{CloudPlatform, DatasetSummaryModel, SnapshotModel}
+
+import scala.jdk.CollectionConverters.CollectionHasAsScala
+
+// A basic wrapper class to simplify interaction with the SnapshotModel from TDR
+class WrappedSnapshot(snapshot: SnapshotModel) {
+
+  // While getSource returns a list of SnapshotSourceModels, this is an artifact and we can make an assumption that
+  // a given snapshot will have exactly one source that will always have a dataset.  Violations of this assumption
+  // indicate a bug, in which case, crash and burn.
+  private def snapshotDataset(): DatasetSummaryModel = getOnlyElement(snapshot.getSource.asScala.toList).getDataset
```

Comment:
is this true, that snapshots expect to have only one source dataset? If that's the case, it might simplify code in Import Service and Terra UI too.

Reply:
Yes, from discussing this with @nmalfroy, there will only ever be one `source` dataset, so while it's a List, it'll always be a singleton.  And the `cloudPlatform` should always be set on it.

If TDR ever _changes_ this assumption, as in, make it something other than exactly one source, they'll do so loudly!